src/main/scala/com/databricks/spark/redshift/RedshiftWriter.scala

```diff
@@ -180,6 +180,12 @@ class RedshiftWriter(jdbcWrapper: JDBCWrapper) extends Logging {
   def saveToRedshift(sqlContext: SQLContext, data: DataFrame, params: MergedParameters) : Unit = {
     val conn = jdbcWrapper.getConnector(params.jdbcDriver, params.jdbcUrl, new Properties()).apply()
 
+    if (params.avrocompression != null && params.avrocompression.nonEmpty) {
+      val conf = sqlContext.sparkContext.hadoopConfiguration
```

Comment:
Instead of mutating the (effectively) global Hadoop configuration, I think that it would be better to clone this configuration and pass our own local copy throughout the Redshift library.  I'm going to make a separate pull request to perform this cleanup.